### ISSUE TITLE
`unicorn/prefer-top-level-await` False positive in svelte file

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#3488]
So I have something along these lines:

```svelte
<script lang="ts">
	async function loadCached(): Promise<string> {
		return fetch(...);
	}
	let hotfixesPromise = $state(loadCached());
</script>

{#await hotfixesPromise}
	fetching...
{:then hotfixes}
	<p><textarea value={hotfixes}></textarea></p>
	<button type="button" onclick={() => (hotfixesPromise = fetchHotfixes())}>Fetch hotfixes</button>
{/await}
```

And the rule is reporting this line:

```ts
	let hotfixesPromise = $state(loadCached());
```

I think that the rule should be disabled in svelte files since this is already how "top level await" works in the template.

This is part of my eslint config

```js
export default defineConfig([
	...eslintConfigXo({
		browser: true,
	}),
	{
		files: ['**/*.svelte'],
		extends: [sveltePlugin.configs['flat/recommended']],
		languageOptions: {
			parserOptions: {
				parser: '@typescript-eslint/parser',
			},
		},
	},
```

**Comment by fregante** [comment]
It's also auto-fixed to this:

```js
	let hotfixesPromise = $state(await loadCached());
```

and then svelte throws unless an experimental flag is enabled

```
Cannot use `await` in deriveds and template expressions, or at the top level of a component, unless the `experimental.async` compiler option is `true`
https://svelte.dev/docs/svelte/await-expressions
```

but still it changes the behavior and intent of the code even if I were to enable the flag.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/pull/3489
